### PR TITLE
Modest refactoring of the migration stages code.

### PR DESCRIPTION
**Pull request by davebx**
The previous code was iterating over the same list twice. This _may_ fix the test failure on #7246

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/controllers/admin.py
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
            if not item.endswith('.py') or item.startswith('0001_tools'):
```

**Comment by jmchilton** [comment]
Marking as WIP and pushing because waiting on a comment response.

**Comment by davebx** [comment]
Thanks for the reminder, @jmchilton.

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
@davebx I've rebased to have CircleCI tests run, will merge if everything is green.

**Comment by davebx** [comment]
Cool, thanks for the help, @nsoranzo and @bgruening

**Pull request merged**
